lucille-core/src/test/java/com/kmwllc/lucille/stage/ParseDateTest.java

```diff
@@ -52,7 +59,119 @@ public void testGetLegalProperties() throws StageException {
             "source",
             "dest",
             "conditions",
-            "class"),
+            "class",
+            "time_zone_id"),
         stage.getLegalProperties());
   }
+
+  @Test
+  public void testInvalidTimeZoneId() {
+    Throwable exception = assertThrows(StageException.class, () -> factory.get("ParseDateTest/invalidTimeZoneConfig.conf"));
+    Throwable cause = exception.getCause().getCause();
+    assertEquals(IllegalArgumentException.class, cause.getClass());
+    assertEquals("Invalid time zone ID: \"invalid\", must be one of TimeZone.getAvailableIDs()", cause.getMessage());
+  }
+
+  @Test
+  public void testParseDateWithTimeZone() throws StageException {
+
+    // create a test config without a time zone
+    HashMap<String,Object> configValues = new HashMap<>();
+    configValues.put("class", "com.kmwllc.lucille.stage.ParseDate");
+    configValues.put("source", List.of("date"));
+    configValues.put("dest", List.of("output"));
+    configValues.put("format_strs", List.of("yyyy-MM-dd"));
```

Comment:
We should test two cases:

- if we the date we're parsing doesn't include zone information, then time_zone_id is used
- if the date we're parsing does include zone information, then that info is used instead of time_zone_id

This test could be updated to use two format_strs like this:

`    configValues.put("format_strs", List.of("yyyy-MM-dd z", "yyyy-MM-dd"));`

When running the Stage, we should try it on two different dates:

- "2021-02-02"
- "2021-02-02 EST"

and make sure we get back the right date (in Zulu format) in each case

this could also be split into a separate test